src/main/java/reactor/ipc/netty/NettyOutbound.java

```diff
@@ -308,8 +313,12 @@ default NettyOutbound sendObject(Publisher<?> dataStream) {
 	 * any error during write
 	 */
 	default NettyOutbound sendObject(Object msg) {
-		return then(FutureMono.deferFuture(() -> context().channel()
-		                                                  .writeAndFlush(msg)));
+		return then(FutureMono.deferFuture((sctx) -> {
+			Channel channel = context().channel();
```

Comment:
I think its dangerous to use ctx to store the context here, maybe a wrapping PublisherEvent would suit this need better (with source and optional context). Further simplification could try if context is not empty before wrapping.

Reply:
What is `PublisherEvent` my IDEA does not help me in searching for that class. Do you mean `writeAndFlush` something like 

```
class PublisherEvent {
Object source;
Context downstreamContext;
}
```

Reply:
It would be an immutable/ internal static inner final class. You can apply it only if Object is Publisher in addition to only if the context is not empty